spec/requests/api_registrations_spec.rb

```diff
@@ -164,6 +164,49 @@
       registration = Registration.find_by(user_id: user.id, competition_id: favourites_reg.competition.id)
       expect(registration.event_ids.sort).to eq(new_event_ids.sort)
     end
+
+    it 'user gets registration email if they cancel and re-register' do
+      perform_enqueued_jobs do
```

Comment:
You only need to wrap `perform_enqueued_jobs` around those parts of your test that actually enqueue the job. So in this case, it would probably be the `patch` below.

Reply:
Another option is to just write your test as usual, and call `perform_enqueued_jobs` directly as a method to "flush" the job queue, without passing a `do` block

Reply:
Done